docs/ml-features.md

```diff
@@ -855,6 +855,58 @@ for more details on the API.
 
 </div>
 
+## TargetEncoder
+
+Target Encoding maps a column of categorical indices into a numerical feature derived from the target.
+Leveraging the relationship between categorical features and the target variable, target encoding usually performs better than one-hot encoding (while avoiding the need to add extra columns)
+
+`TargetEncoder` can transform multiple columns, returning a single target-encoded output column for each input column.
+User can specify input and output column names by setting `inputCol` and `outputCol` for single-column use cases, or `inputCols` and `outputCols` for multi-column use cases (both arrays required to have the same size)
+
+User can specify the target column name by setting `label`. This column contains the ground-truth labels from which encodings will be derived
+
+`TargetEncoder` supports the `handleInvalid` parameter to choose how to handle invalid input during transforming data.
+Available options include 'keep' (any invalid inputs are assigned to an extra categorical index) and 'error' (throw an error).
+
+`TargetEncoder` supports the `targetType` parameter to choose the label type when fitting data, affecting how statistics are calculated.
+Available options include 'binary'  and 'continuous' (mean-encoding).
+When set to 'binary', encodings will be fitted from target conditional probabilities (a.k.a bin-counting).
+When set to 'continuous', encodings will be fitted from according to target mean (a.k.a. mean-encoding).
```

Comment:
I think you can describe this a little bit more somewhere, could be here or at the top - what does target encoding actually do? with a simplistic example of a few rows?

Just want to make it immediate clearly in 1 paragraph what this is doing for binary vs continuous targets

Reply:
done